src/mainFilesSidebarLoader.js

```diff
@@ -20,50 +20,80 @@
  *
  */
 
-import './init.js'
-import FilesSidebarCallView from './views/FilesSidebarCallView.js'
+import Vue from 'vue'
+
+import { getRequestToken } from '@nextcloud/auth'
+import { generateFilePath } from '@nextcloud/router'
+
+import FilesSidebarTabLoader from './FilesSidebarTabLoader.vue'
+
+// CSP config for webpack dynamic chunk loading
+// eslint-disable-next-line
+__webpack_nonce__ = btoa(getRequestToken())
+
+// Correct the root of the app for chunk loading
+// OC.linkTo matches the apps folders
+// OC.generateUrl ensure the index.php (or not)
+// We do not want the index.php since we're loading files
+// eslint-disable-next-line
+__webpack_public_path__ = generateFilePath('spreed', '', 'js/')
+
+Vue.prototype.OC = OC
+Vue.prototype.OCA = OCA
+
+const loaderTab = () => new Vue({
+	id: 'talk-chat-tab',
+	render: h => h(FilesSidebarTabLoader),
+})
 
 const isEnabled = function(fileInfo) {
 	if (fileInfo && !fileInfo.isDirectory()) {
 		return true
 	}
 
-	const token = OCA.Talk.store.getters.getToken()
+	const token = OCA.Talk.store?.getters.getToken()
 
 	// If the Talk tab can not be displayed then the current conversation is
 	// left; this must be done here because "setFileInfo" will not get
 	// called with the new file if the tab can not be displayed.
 	if (token) {
-		OCA.Talk.store.dispatch('leaveConversation', { token })
+		OCA.Talk.store?.dispatch('leaveConversation', { token })
 	}
 
-	OCA.Talk.store.dispatch('updateTokenAndFileIdForToken', {
+	OCA.Talk.store?.dispatch('updateTokenAndFileIdForToken', {
 		newToken: null,
 		newFileId: null,
 	})
 
 	return false
 }
 
-// It might be enough to keep the instance only in the Tab object itself,
-// without using a shared variable that can be destroyed if a new tab is
-// mounted and the previous one was not destroyed yet, as the tabs seem to
-// always be properly destroyed. However, this is how it is done for tabs in
-// server, so it is done here too just to be safe.
-let tabInstance = null
+if (!window.OCA.Talk) {
+	window.OCA.Talk = {}
+}
+Object.assign(window.OCA.Talk, {
+	fileInfo: null,
+	loaderTab,
+	isFirstLoad: true,
+	// It might be enough to keep the instance only in the Tab object itself,
+	// without using a shared variable that can be destroyed if a new tab is
+	// mounted and the previous one was not destroyed yet, as the tabs seem to
+	// always be properly destroyed. However, this is how it is done for tabs in
+	// server, so it is done here too just to be safe.
+	tabInstance: null,
```

Comment:
Do we need the tab instance as a part of the public API in OCA.Talk?

Reply:
not neccessary to have it in the public, but this variable should be shared between mainFilesSidebar and mainFilesSidebarLoader, so Sidebar would work correctly with the reference (mount, update, destroy)

Reply:
I believe, it should work as expected?
```js
const PrivateTalk = {
	tabInstance: null,
}
export default PrivateTalk
```